process/peer/process.go

```diff
@@ -954,21 +1004,30 @@ func (vs *validatorStatistics) display(validatorKey string) {
 	)
 }
 
-func (vs *validatorStatistics) decreaseAll(shardID uint32, missedRounds uint64, epoch uint32) error {
+func (vs *validatorStatistics) decreaseAll(
+	shardID uint32,
+	missedRounds uint64,
+	epoch uint32,
+) error {
+	if epoch < vs.ratingEnableEpoch {
+		return nil
+	}
 
 	log.Trace("ValidatorStatistics decreasing all", "shardID", shardID, "missedRounds", missedRounds)
 	consensusGroupSize := vs.nodesCoordinator.ConsensusGroupSize(shardID)
 	validators, err := vs.nodesCoordinator.GetAllEligibleValidatorsPublicKeys(epoch)
 	if err != nil {
 		return err
 	}
+
 	shardValidators := validators[shardID]
 	validatorsCount := len(shardValidators)
 	percentageRoundMissedFromTotalValidators := float64(missedRounds) / float64(validatorsCount)
 	leaderAppearances := uint32(percentageRoundMissedFromTotalValidators + 1 - math.SmallestNonzeroFloat64)
 	consensusGroupAppearances := uint32(float64(consensusGroupSize)*percentageRoundMissedFromTotalValidators +
 		1 - math.SmallestNonzeroFloat64)
 	ratingDifference := uint32(0)
+
```

Comment:
So basically, won't decrease the rating if a consensus round is missed but we allow to increase the ratings on all peers (during the epoch grace period).

Reply:
those who are working well, should be rewarded as well.